cdap-support-bundle/bin/observability.py

```diff
@@ -0,0 +1,507 @@
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/system/services/status -H "{auth_token}"',
+            "fileName": 'kubectl-exec-system-service-status',
+            "fileExt": 'txt',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_status
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/namespaces/system/apps/pipeline/services/studio/status -H "{auth_token}"',
+            "fileName": 'kubectl-exec-system-service-studio-status',
+            "fileExt": 'txt',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_status
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/namespaces/system/apps/dataprep/services/service/status -H "{auth_token}"',
+            "fileName": 'kubectl-exec-system-service-wrangler-status',
+            "fileExt": 'txt',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_status
+        }
+    ]
+
+    log_commands_processor(cdapCommandLists, directoryName, True, cdap_ns, pipeline_name, run_id)
+
+    update_progress(98)
+    compressed_file = shutil.make_archive(
+        base_name=directoryName,
+        format='gztar',
+        root_dir=parent_dir + directoryName
+    )
+
+    shutil.rmtree(directoryName)
+    update_progress(100)
+    print(f'Logs collection completed. Please find {directoryName}.tar.gz in current directory')
+    exit()
+
+
+def log_file_write(pathName, filename, fileWriteMode, contentToWrite, fileExt="log"):
+    f = open(f"{pathName}/{filename}.{fileExt}", fileWriteMode)
+    f.write(contentToWrite)
+    f.close()
+
+
+def update_progress(count):
+    total = 100
+    suffix = 'Done' if (count == 100) else 'In Progress'
+    bar_len = 60
+    filled_len = int(round(bar_len * count / float(total)))
+
+    percents = round(100.0 * count / float(total), 1)
+    bar = '=' * filled_len + '-' * (bar_len - filled_len)
+
+    sys.stdout.write('[%s] %s%s ...%s\r' % (bar, percents, '%', suffix))
+    sys.stdout.flush()
+
+
+def log_commands_processor(commandLists, directoryName, progressShow, cdap_ns, pipeline_name, run_id):
+    progress_len = 5
+    count = 1
+
+    for x in commandLists:
+        directory_name = directoryName
+        if "isPipelineLog" in x  and (cdap_ns is None or pipeline_name is None or run_id is None):
```

Comment:
looks like we're not really using run_id? If the user provides a run_id, we should collect logs for that run. Otherwise collect data about CDAP system services. See this comment - https://github.com/cdapio/cdap/pull/14867#pullrequestreview-1264089479

Reply:
We will use it for pipeline logs but currently we are not able to see the deployed pipelines on HDF. Waiting for that to fix.

Reply:
As discussed offline, this script won't be able to collect pipeline logs for tethered runs as the logs will go to CDF.
However, we should collect logs for non-tethered runs

Reply:
Additionally, we should also capture logs/kubectl command outputs from the program runtime namespace


kubectl get jobs -n $runtimeNamespace
kubectl get configmap -n $runtimeNamespace
kubectl get pods -l cdap.twill.run.id=$RUN_ID -n $runtimeNamespace>-o wide
Logs from pods listed in output of previous command
kubectl describe resources listed above